operator/es_client.go

```diff
@@ -350,29 +351,37 @@ func (esSettings *ESSettings) updateRebalance(value string) {
 // repeatedly query shard allocations to ensure success of drain operation.
 func (c *ESClient) waitForEmptyEsNode(ctx context.Context, pod *v1.Pod) error {
 	podIP := pod.Status.PodIP
-	_, err := resty.New().
+
+	// Counter to track the number of retries
+	retryCount := 0
+
+	_, err := resty.NewWithClient(&http.Client{Transport: http.DefaultTransport}).
 		SetRetryCount(c.DrainingConfig.MaxRetries).
```

Comment:
The default max retries is 999 which is a bit too much. Let's reduce it to a more fail fast strategy to avoid elongated  loop of retries. Even if the current loop fails, the operator must ideally pick up where it left off.

Reply:
it is configurable right? We can lower this for our own deployment.